benchmarks/ALCE/ELI5/eli5.py

```diff
@@ -0,0 +1,176 @@
+import json
+import os
+import time
+
+from datasets import Dataset
+from tqdm import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+import rageval as rl
+from rageval.models.openai import OpenAILLM
+from utils import make_demo, create_eli5_eval_dataset
+
+
+class ELI5:
+
+    def __init__(self, args):
+        self.args = args
+
+        self.model = None
+        self.tokenizer = None
+        self.eval_data = None
+        self.prompt = None
+        self.name = None
+        self.result_path = None
+
+    def generate_init(self):
+        print("-" * 10 + "Loading model" + "-" * 10)
+        if "gpt-3.5-turbo" in self.args.model:
+            os.environ["OPENAI_API_KEY"] = self.args.api_key
+            self.model = OpenAILLM(
```

Comment:
Now the benchmark can be run in the following way
```
eli5 = ELI5(args)

# in case we need to predict the result with some pre-defined model.
eli5.init_model(args.model, args.api_key) # The OpenAI model needs to be passed into the api, but the local model does not need to
gen_result = eli5.predict()
eli5.evaluate()
eli5.save_result(eval_result_path) # Users can specify the save path here. If not specified, it will be saved in the default location.

# in other case where we direct evaluate some existing result
eli5.evaluate(gen_result_path) # The user needs to specify an existing generated result file.
eli5.save_result(eval_result_path) # Users can specify the save path here. If not specified, it will be saved in the default location.
```

`model = OpenAILLM()` may not be necessary in order to keep the OpenAI model and the local model using the same process to execute the benchmark as much as possible.